src/PrestaShopBundle/Controller/Api/OAuth2/AccessTokenController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Controller\Api\OAuth2;
+
+use League\OAuth2\Server\AuthorizationServer;
```

Comment:
There is no use of the interface here. This controller is for providing an Authorization Server while the interface is for the Resource Server to be able to check the token validity.

Reply:
I agree there is no interface here, but you have created an interface to "hide" League logic in the oauth server, but the same thing should be done here. Otherwise the other interface is not needed I think.

Reply:
The other interface is needed for modules that want to use their own Authorization Server, you can have a look at the module I added in the "how to test" section ;)